ai_edge_torch/generative/layers/experimental/kv_cache.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2024 The AI Edge Torch Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+# Cache management utilities.
+
+from typing import Any, Dict, List, Tuple
+
+import torch
+
+
+class KVCache:
+  """A utility clss for holding Key and Value caches per layer."""
+
+  def __init__(
+      self,
+      num_layers: int,
+      shape: Tuple[int],
+      dtype: torch.dtype = torch.float32,
+      device: torch.device = None,
```

Comment:
is this useful for our use case?

Reply:
It's generally good practice to allocate on the correct device instead of moving later. Beside, it has a default and should not matter if not needed.